components/proxy/src/main/java/com/hotels/styx/proxy/backends/file/BackendServiceDeserializer.java

```diff
@@ -0,0 +1,295 @@
+            this.tlsSettings = tlsSettings;
+            return this;
+        }
+
+        /**
+         * Sets hosts.
+         *
+         * @param origins origins
+         * @return this builder
+         */
+        public Builder origins(Origin... origins) {
+            return origins(ImmutableSet.copyOf(origins));
+        }
+
+        /**
+         * Sets rewrites to be performed on URLs.
+         *
+         * @param rewriteConfigs rewrite configuration
+         * @return this builder
+         */
+        public Builder rewrites(RewriteConfig... rewriteConfigs) {
+            return rewrites(asList(rewriteConfigs));
+        }
+
+        /**
+         * Sets rewrites to be performed on URLs.
+         *
+         * @param rewriteConfigs rewrite configuration
+         * @return this builder
+         */
+        @JsonProperty("rewrites")
+        public Builder rewrites(List<RewriteConfig> rewriteConfigs) {
+            this.rewrites = ImmutableList.copyOf(rewriteConfigs);
+            return this;
+        }
+
+        /**
+         * Sets connection pool configuration.
+         *
+         * @param connectionPoolSettings connection pool configuration
+         * @return this builder
+         */
+        @JsonProperty("connectionPool")
+        public Builder connectionPoolConfig(ConnectionPoolSettings connectionPoolSettings) {
+            this.connectionPoolSettings = checkNotNull(connectionPoolSettings);
+            return this;
+        }
+
+        /**
+         * Sets connection pool configuration.
+         *
+         * @param connectionPoolSettings connection pool configuration
+         * @return this builder
+         */
+        public Builder connectionPoolConfig(ConnectionPool.Settings connectionPoolSettings) {
+            this.connectionPoolSettings = checkNotNull(connectionPoolSettings);
+            return this;
+        }
+
+        /**
+         * Sets sticky-session configuration.
+         *
+         * @param stickySessionConfig sticky-session configuration.
+         * @return this builder
+         */
+        @JsonProperty("stickySession")
+        public Builder stickySessionConfig(StickySessionConfig stickySessionConfig) {
+            this.stickySessionConfig = checkNotNull(stickySessionConfig);
+            return this;
+        }
+
+        /**
+         * Sets health-check configuration.
+         *
+         * @param healthCheckConfig health-check configuration
+         * @return this builder
+         */
+        @JsonProperty("healthCheck")
+        public Builder healthCheckConfig(HealthCheckConfig healthCheckConfig) {
+            this.healthCheckConfig = checkNotNull(healthCheckConfig);
+            return this;
+        }
+
+        /**
+         * Builds the application.
+         *
+         * @return the application
+         */
+        public BackendServiceDeserializer build() {
+            BackendService.Builder builder = BackendService.newBackendServiceBuilder();
```

Comment:
Builder interfaces are fluent. There is no need to repeat `builder.`
This code could be:
```
 public BackendServiceDeserializer build() {
            BackendService.Builder builder = BackendService.newBackendServiceBuilder()
                    .id(this.id)
                    .path(this.path)
                    .origins(this.origins)
                    .connectionPoolConfig(this.connectionPoolSettings)
                    .healthCheckConfig(this.healthCheckConfig)
                    .stickySessionConfig(this.stickySessionConfig)
                    .rewrites(this.rewrites)
                    .responseTimeoutMillis(this.responseTimeoutMillis)
                    .https(this.tlsSettings)
                    .httpsOld(this.tlsSettings);
            return new BackendServiceDeserializer(builder);
        }
```